core/resources.rs

```diff
@@ -134,6 +134,10 @@ impl ResourceTable {
   /// Returns a unique resource ID, which acts as a key for this resource.
   pub fn add_rc<T: Resource>(&mut self, resource: Rc<T>) -> ResourceId {
     let resource = resource as Rc<dyn Resource>;
+    self.add_rc_dyn(resource)
+  }
+
+  pub fn add_rc_dyn(&mut self, resource: Rc<dyn Resource>) -> ResourceId {
```

Comment:
Why is this API needed?

Reply:
Structs (e.g. `SqliteBackedCache`) that implement the `Cache` trait return dynamic resources, so I wanted this API to add them to the state.